spectrum-network/tests/integration_tests/aggregation.rs

```
@@ -0,0 +1,238 @@
+};
+use spectrum_network::protocol_handler::handel::{HandelConfig, Threshold};
+use spectrum_network::protocol_handler::sigma_aggregation::{types::PublicKey, SigmaAggregation};
+use spectrum_network::protocol_handler::ProtocolHandler;
+use spectrum_network::types::{ProtocolVer, Reputation};
+
+pub struct Peer {
+    pub peer_id: PeerId,
+    pub peer_addr: Multiaddr,
+    pub peer_pk: k256::PublicKey,
+    pub peer_handle: AbortHandle,
+    pub aggr_handler_mailbox: Sender<AggregationAction<Blake2b>>,
+}
+
+pub fn k256_to_libsecp256k1(secret_key: SecretKey) -> identity::secp256k1::SecretKey {
+    identity::secp256k1::SecretKey::try_from_bytes(secret_key.to_bytes().as_mut_slice()).unwrap()
+}
+
+pub fn setup_nodes(n: usize) -> Vec<Peer> {
+    let mut rng = OsRng;
+    let mut spawn_node = move |node_ix| {
+        let peer_sk = SecretKey::random(&mut rng);
+        let peer_key = identity::Keypair::from(identity::secp256k1::Keypair::from(k256_to_libsecp256k1(
+            peer_sk.clone(),
+        )));
+        use elliptic_curve::sec1::ToEncodedPoint;
+        let k256_pk = peer_sk.public_key();
+        let k256_point = k256_pk.to_encoded_point(true);
+        let k256_encoded = k256_point.as_bytes();
+        let libp2p_pk = libp2p_identity::secp256k1::PublicKey::decode(k256_encoded).unwrap();
+        let peer_id = PeerId::from_public_key(&libp2p_identity::PublicKey::Secp256k1(libp2p_pk));
+        let other_peer_id = PeerId::from(peer_key.public());
+        assert_eq!(peer_id, other_peer_id);
+        let peer_addr: Multiaddr = format!("/ip4/127.0.0.1/tcp/{}", 8000 + node_ix).parse().unwrap();
+
+        let bb = peer_sk.to_bytes().to_vec();
+        let key = SecretKey::from_slice(&bb).unwrap();
+        assert_eq!(key, peer_sk);
+
+        //generate_peer_info_files(n);
+
+        let one_shot_proto_conf = OneShotProtocolConfig {
+            version: ProtocolVer::default(),
+            spec: OneShotProtocolSpec {
+                max_message_size: 5000,
```

If this is too small, node would silently drop transmission of one-shot messages.